core/domain/exp_services.py

```diff
@@ -1958,16 +2147,21 @@ def are_changes_mergeable(exp_id, frontend_version, change_list):
                 if change.state_name in state_names_of_renamed_states:
                     old_state_name = (
                         state_names_of_renamed_states[change.state_name])
-                    new_state_name = (
-                        state_names_of_renamed_states[change.state_name])
-                if change.state_name in old_to_new_state_names:
-                    new_state_name = old_to_new_state_names[old_state_name]
+                if old_state_name in old_to_new_state_names:
+                    if feconf.CAN_SEND_EMAILS:
+                        email_manager.send_mail_to_admin(
+                            'Adding/Deleting State Changes not mergeable',
+                            'The exploration id is: %s, \n '
+                            'The frontend version is: %s \n '
+                            'The changes list is: %s '
+                            % (exp_id, frontend_version, change_list))
+                    return False
```

Comment:
Can you please add a comment about why this is required i.e. why we want to catch these errors?

Reply:
Done.